### PR TITLE
feat: creation of TimesheetCard

### PR DESCRIPTION
**Pull request by vhu-axelor**
RM#71325

### CODE REVIEW

**Review by gca-axelor — APPROVED** [review]

File: packages/apps/hr/src/i18n/en.json [hunk 1]
```diff
@@ -45,11 +45,8 @@
   "Hr_TimeUnit_Days": "d",
   "Hr_TimeUnit_Hours": "h",
   "Hr_TimeUnit_Minutes": "min",
-  "Hr_SliceAction_FetchExpenseLines": "fetch expense lines",
-  "Hr_SliceAction_SearchDraftExpense": "fetch draft expenses",
-  "Hr_SliceAction_FetchExpense": "fetch expenses",
-  "Hr_SliceAction_FetchMyExpense": "fetch my expense",
-  "Hr_SliceAction_FetchExpenseToValidate": "fetch expense to validate",
```
Comment (gca-axelor) [inline packages/apps/hr/src/i18n/en.json:52]: why ?????????????

Comment (vhu-axelor) [reply packages/apps/hr/src/i18n/en.json]: This was not intended, this must have happened when I added the new translations

File: packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        isCompleted={isCompleted}
+        startDate={startDate}
+        endDate={endDate}
+        company={company}
+        totalDuration={totalDuration}
+        style={styles.cardContainer}
+        onPress={onPress}
+      />
+      {isActions && (
+        <View style={styles.iconContainer}>
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx:63]: Remove wrapping view

File: packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import {StyleSheet, View} from 'react-native';
+import {CardIconButton, useThemeColor} from '@axelor/aos-mobile-ui';
+import {TimesheetCard} from '../../atoms';
+
+interface TimesheetDetailCardProps {
+  isCompleted: boolean;
+  startDate: string;
+  endDate: string;
+  company: string;
+  totalDuration: string;
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx:29]: wrong type

File: packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    flexDirection: 'row',
+    alignSelf: 'center',
+    marginVertical: 2,
+  },
+  cardContainer: {
+    flex: 6,
+  },
+  iconContainer: {
+    flex: 1,
+  },
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx:88]: won't be needed

File: packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  cardIconButton: {
+    flex: 1,
+    marginLeft: 5,
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/molecules/TimesheetDetailCard/TimesheetDetailCard.tsx:91]: Is marginLeft really necessary ?

File: packages/apps/hr/src/i18n/en.json [hunk 6]
```diff
@@ -45,6 +45,8 @@
   "Hr_TimeUnit_Days": "d",
   "Hr_TimeUnit_Hours": "h",
   "Hr_TimeUnit_Minutes": "min",
+  "Hr_Company": "Company",
```
Comment (lme-axelor) [inline packages/apps/hr/src/i18n/en.json:48]: Use base translation

File: packages/apps/hr/src/i18n/fr.json [hunk 7]
```diff
@@ -45,6 +45,8 @@
   "Hr_TimeUnit_Days": "j",
   "Hr_TimeUnit_Hours": "h",
   "Hr_TimeUnit_Minutes": "min",
+  "Hr_Company": "Entreprise",
```
Comment (lme-axelor) [inline packages/apps/hr/src/i18n/fr.json:48]: Use base translation

**Comment by vhu-axelor** [comment]
To test it you can put this code in ExpenseListScreen.js:
`/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {Screen} from '@axelor/aos-mobile-ui';
import {TimesheetDetailCard} from '../components';

const ExpenseListScreen = ({}) => {
  const demoData = [
    {
      isCompleted: false,
      startDate: '2000-10-20',
      endDate: '2023-10-20',
      company: 'Entreprise',
      totalDuration: 3,
      onPress: () => console.log('OK'),
    },
    {
      isCompleted: true,
      startDate: '2000-10-20',
      endDate: '2023-10-20',
      company: 'Entreprise avec un nom super long',
      totalDuration: 3.5,
      onPress: () => console.log('OK'),
    },
  ];

  return (
    <Screen>
      {demoData.map((item, index) => {
        return (
          <TimesheetDetailCard
            isCompleted={item.isCompleted}
            startDate={item.startDate}
            endDate={item.endDate}
            company={item.company}
            totalDuration={item.totalDuration}
            onPress={item.onPress}
            key={index}
          />
        );
      })}
    </Screen>
  );
};

export default ExpenseListScreen;
`

**Pull request merged**